src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,173 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    static class Node<T> {
+
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(T data) {
+            this.value = data;
+            this.prev = null;
+            this.next = null;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (head == null) {
+            head = newNode;
+            tail = newNode;
+
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
```

Comment:
Consider creating a `checkIndex` method to verify if the index is valid and use it in methods like `add`, `get`, `set`, and `remove` to avoid code duplication.

Reply:
ok, done, but in add method, checking the index is a little bit different...